src/ui/components/SearchInput/index.js

```diff
@@ -0,0 +1,88 @@
+/* global document, window */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+import Icon from 'ui/components/Icon';
+
+import './style.scss';
+
+export default class SearchInput extends React.Component {
+  static propTypes = {
+    className: PropTypes.string,
+    defaultValue: PropTypes.string,
+    name: PropTypes.string.isRequired,
+    placeholder: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { focus: false, value: props.defaultValue };
+  }
+
+  componentDidMount() {
+    this.setIconPosition();
+    window.addEventListener('resize', this.setIconPosition);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.setIconPosition);
+  }
+
+  onBlur = () => {
+    this.setState({ focus: false });
+  }
+
+  onFocus = () => {
+    this.setState({ focus: true });
+  }
+
+  onInput = (e) => {
+    this.setState({ value: e.target.value });
+  }
+
+  onMouseDown = (e) => {
+    this.setState({ focus: true });
+    if (!this.input.value) {
+      e.preventDefault();
+      const setFocus = () => {
+        this.input.focus();
+        this.animateIcon.removeEventListener('transitionend', setFocus);
+      };
+      this.animateIcon.addEventListener('transitionend', setFocus);
+    }
+  }
+
+  setIconPosition = () => {
+    const { left: labelLeft } = this.labelIcon.getBoundingClientRect();
+    const { left: animateLeft } = this.animateIcon.getBoundingClientRect();
+    this.animateIcon.style.transform = `translateX(${labelLeft - animateLeft}px)`;
```

Comment:
I believe so but I'll double check. I remember trying to figure out how to handle the RTL case and then I think avoiding it entirely and doing the difference of x positions worked in both cases because `translateX()` is always from the left.